Marlin/temperature.cpp

```diff
@@ -1544,70 +1546,79 @@ void Temperature::isr() {
   #endif
 
   #if DISABLED(SLOW_PWM_HEATERS)
```

Comment:
The preprocessor is always run! "ENABLED(...)" by itself *is* a preprocessor expression, which is expanded prior to the compilation itself.

If you want to see the output of the preprocessor, pass "-E" to gcc.

I tried it, checked the created preprocessor output, and checked the generated assembly. If it does not work for you, whats the error message?